ci/jenkins/utils.sh

```diff
@@ -14,6 +14,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+function check_and_cleanup_buildkit_cache() {
+    free_space_threshold=40
+    
+    docker builder prune -af --keep-storage=10gb > /dev/null
```

Comment:
I hope that in the general case this is enough and we can keep some cache around to speed up future builds

Reply:
maybe we should start with the size one actually. I imagine that the implementation is "smart" enough to delete old objects first. Otherwise, if we don't trigger a job for 1h, we lose all the cache and it becomes useless.
if you prefer, you can go back to your first version:
```
if free_space < threshold
    docker builder prune -af --keep-storage=10gb > /dev/null
if free_space < threshold
    docker builder prune -af > /dev/null
```
I think the size filter is a better fit for us (for the build cache)

Reply:
Sure, I am okay with it.